receiver/nginxreceiver/internal/record/recorder.go

```diff
@@ -0,0 +1,55 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package record
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/nginxreceiver/internal/metadata"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/nginxreceiver/internal/model"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+)
+
+// Item extracts data from NGINX Access Items and records them using the given MetricsBuilder.
+func Item(ai *model.NginxAccessItem, mb *metadata.MetricsBuilder) error {
+	now := pcommon.NewTimestampFromTime(time.Now())
+
+	if ai.Status != "" {
+		codeRange, err := mapCodeRange(ai.Status)
+		if err != nil {
+			return fmt.Errorf("code range parse: %w", err)
+		}
+
+		mb.RecordNginxHTTPStatusDataPoint(now, 1, codeRange)
+	}
+
+	return nil
+}
+
+func mapCodeRange(statusCode string) (metadata.AttributeStatusCode, error) {
+	number, err := strconv.Atoi(statusCode)
+	if err != nil {
+		return 0, fmt.Errorf("cast status code to int: %w", err)
+	}
+
+	// We want to "floor" the response code, so we can map it to the correct range (i.e. to 1xx, 2xx, 4xx or 5xx).
+	codeRange := (number / 100) * 100
```

Comment:
Is https://pkg.go.dev/math#Floor useable here?

Reply:
Nope, don't think so. `Floor` takes in a float and our `number` is an int. The calculation performed here works just fine for what it's trying to achieve, as integer division automatically rounds down the result.